pkg/controller/sdk/finalizers_accessor.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sdk
+
+import (
+	"reflect"
+)
+
+// FinalizersAccessor is the interface for a Resource that implements the getter and setting for
+// accessing its Finalizer set.
+// +k8s:deepcopy-gen=true
+type FinalizersAccessor interface {
+	GetFinalizers() interface{}
```

Comment:
or put another way, why not:
 GetFinalizers() []string
or what I've seen typically happen is using the Set abstraction for these.

Reply:
k8s.io/apimachinery/pkg/util/sets

Reply:
Switched to using `sets.String`. It isn't useful for this change, but if we want to manipulate Finalizers in the future, it will be much easier.